### PR TITLE
Enh/generate lfp

### PR DESCRIPTION
**Pull request by mdenker**
This PR closes # #594.

### Changes
* add example to doc-string
* handle one-dimensional arrays as input for `x_positions`
* add regression unit-tests

### CODE REVIEW

File: elephant/current_source_density.py [hunk 1]
```diff
@@ -227,7 +227,7 @@ def generate_lfp(csd_profile, x_positions, y_positions=None, z_positions=None,
         1D : gauss_1d_dipole
         2D : large_source_2D and small_source_2D
         3D : gauss_3d_dipole
-    x_positions : np.ndarray
+    x_positions : np.ndarray # TODO: add dimensionality for x-y-z_positions
```
Comment (Kleinjohann) [inline elephant/current_source_density.py:230]: Is this to be addressed in ths PR or another one?

File: elephant/test/test_current_source_density.py [hunk 2]
```diff
@@ -153,5 +154,29 @@ def test_kcsd2d_init(self):
...
+        cls.one_dimensional = np.linspace(0, 10, 2304)
+        cls.two_dimensional = np.linspace(0, 10, 2304
+                                          ).reshape(2304, 1)
+
+    def test_generate_lfp_one_dimensional_array(self):
+        """
+        Regression test for Issue #546,
+        see: https://github.com/NeuralEnsemble/elephant/issues/546
+        """
+        # this should raise an error
```
Comment (Kleinjohann) [inline elephant/test/test_current_source_density.py]: hopefully not any more?
```suggestion
        # this should raise NOT an error
```

File: elephant/test/test_current_source_density.py [hunk 3]
```diff
@@ -153,5 +154,29 @@ def test_kcsd2d_init(self):
...
+        generate_lfp(utils.gauss_1d_dipole, self.one_dimensional)
+
+    def test_generate_lfp_two_dimensional_array(self):
+        """
+        Regression test for Issue #546,
+        see: https://github.com/NeuralEnsemble/elephant/issues/546
+        """
+        # this should raise an error
```
Comment (Kleinjohann) [inline elephant/test/test_current_source_density.py]:
```suggestion
        # this should NOT raise an error
```
see anpve

**Review by Kleinjohann — APPROVED** [review]
Looks good, suggested some minor changes.

File: elephant/current_source_density.py (hunk 1 above)
Comment (mdenker) [reply elephant/current_source_density.py:230]: True, I'll do it later...

**Comment by Moritz-Alexander-Kern** [comment]
This was used as a demo for the INM-Retreat 2023 tutorial: 

"Research Data Management and Research Software Engineering"

See also: https://events.hifis.net/event/888/page/206-program-17-october-2023

**Pull request closed without merging**